src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorSelector.java

```diff
@@ -0,0 +1,269 @@
+    }
+
+    @Override
+    public void setMinBitrate(int minBitrateBps) {
+        synchronized (critSect)
+        {
+            ((RemoteBitrateEstimator)packetTransformer)
+                    .setMinBitrate(minBitrateBps);
+            this.minBitrateBps = minBitrateBps;
+        }
+    }
+
+    @Override
+    public RawPacket reverseTransform(
+            RawPacket packet)
+    {
+        /**
+         * @Todo Find out why having synchronized here makes sense since it
+         * exists in PickEstimatorFromHeader.
+         */
+        if (this.absSendTimeEngine == null){
+            this.absSendTimeEngine = stream.getAbsSendTimeEngine();
+        }
+
+        synchronized (critSect)
+        {
+            if(enableUseOfAbsSendTime)
+            {
+                PickEstimatorFromHeader(packet);
+            }
+            packetTransformer.reverseTransform(packet);
+        }
+        return packet;
+    }
+
+    /**
+     * Gets the <tt>PacketTransformer</tt> for RTP packets.
+     *
+     * @return the <tt>PacketTransformer</tt> for RTP packets
+     */
+    @Override
+    public PacketTransformer getRTPTransformer() {
+        return this;
+    }
+
+    /**
+     * Gets the <tt>PacketTransformer</tt> for RTCP packets.
+     *
+     * @return the <tt>PacketTransformer</tt> for RTCP packets
+     */
+    @Override
+    public PacketTransformer getRTCPTransformer() {
+        return null;
+    }
+
+
+    /**
+     * Returns the number of milliseconds until this instance wants a worker
+     * thread to call {@link #run()}. The method is called on the same
+     * worker thread as Process will be called on.
+     *
+     * @return the number of milliseconds until this instance wants a worker
+     * thread to call {@link #run()}
+     */
+    @Override
+    public long getTimeUntilNextRun() {
+        long timeUntilNextRun  =
+                (packetTransformer instanceof RecurringRunnable) ?
+             ((RecurringRunnable) packetTransformer)
+                     .getTimeUntilNextRun() : defaultTimeUntilNextRunMs;
+        return  timeUntilNextRun;
+    }
+
+    /**
+     * When an object implementing interface <code>Runnable</code> is used
+     * to create a thread, starting the thread causes the object's
+     * <code>run</code> method to be called in that separately executing
+     * thread.
+     * <p>
+     * The general contract of the method <code>run</code> is that it may
+     * take any action whatsoever.
+     *
+     * @see Thread#run()
+     */
+    @Override
+    public void run() {
+        if (packetTransformer instanceof Runnable)
+        {
+            ((Runnable)packetTransformer).run();
+        }
+    }
+
+    @Override
+    public void onRttUpdate(long avgRttMs, long maxRttMs) {
+        if(packetTransformer instanceof CallStatsObserver){
+            ((CallStatsObserver)packetTransformer)
+                    .onRttUpdate(avgRttMs,maxRttMs);
```

Comment:
Note that the `RemoteBitrateEstimatorAbsSendTime` does not currently implement the `CallStatsObserver`.